packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,345 @@
+            // uint8 private _initialized;
+            // bool private _initializing;
+            hex"0000000000000000000000000000000000000000000000000000000000000000",
+            // 1
+            // false
+            hex"0000000000000000000000000000000000000000000000000000000000000001"
+        );
+        vm.store(
+            Predeploys.OPTIMISM_MINTABLE_ERC20_FACTORY,
+            // address public bridge;
+            hex"0000000000000000000000000000000000000000000000000000000000000001",
+            // L2_STANDARD_BRIDGE
+            hex"0000000000000000000000004200000000000000000000000000000000000010"
+        );
+
+        vm.etch(address(factory), hex"");
+        vm.resetNonce(address(factory));
+    }
+
+    function _setL1BlockNumber() internal {
+        vm.etch(
+            _predeployToCodeNamespace(Predeploys.L1_BLOCK_NUMBER),
+            vm.getDeployedCode("L1BlockNumber.sol:L1BlockNumber")
+        );
+    }
+
+    function _setGasPriceOracle() internal {
+        vm.etch(
+            _predeployToCodeNamespace(Predeploys.GAS_PRICE_ORACLE),
+            vm.getDeployedCode("GasPriceOracle.sol:GasPriceOracle")
+        );
+    }
+
+    function _setGovernanceToken() internal {
+        if (!cfg.enableGovernance()) {
+            console.log("Governance not enabled, skipping setting governanace token");
+            return;
+        }
+
+        vm.etch(Predeploys.GOVERNANCE_TOKEN, vm.getDeployedCode("GovernanceToken.sol:GovernanceToken"));
+
+        vm.store(
+            Predeploys.GOVERNANCE_TOKEN,
+            // string private _name;
+            hex"0000000000000000000000000000000000000000000000000000000000000003",
+            // "Optimism"
+            hex"4f7074696d69736d000000000000000000000000000000000000000000000010"
+        );
+        vm.store(
+            Predeploys.GOVERNANCE_TOKEN,
+            // string private _symbol;
+            hex"0000000000000000000000000000000000000000000000000000000000000004",
+            // "OP"
+            hex"4f50000000000000000000000000000000000000000000000000000000000004"
+        );
+        vm.store(
+            Predeploys.GOVERNANCE_TOKEN,
+            // address private _owner;
+            hex"000000000000000000000000000000000000000000000000000000000000000a",
+            bytes32(uint256(uint160(cfg.governanceTokenOwner())))
+        );
+    }
+
+    function _setL1Block() internal {
+        vm.etch(
+            _predeployToCodeNamespace(Predeploys.L1_BLOCK_ATTRIBUTES),
+            vm.getDeployedCode("L1Block.sol:L1Block")
+        );
+
+        // TODO - These attributes were set in previous L2 genesis,
+        // do they need to be set here?
+        // vm.prank(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001);
+        // L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).setL1BlockValues(
+        //     0, // TODO Not included in DeployConfig, should it be?
+        //     cfg.l1GenesisBlockTimestamp(),
+        // );
+    }
```

Comment:
Looking at the previously generated L2 genesis, the storage slots for all the L1 block attributes were set, but I'm not sure where that was happening. Do they need to be set from genesis?